rules/react.js

```diff
@@ -1,91 +1,146 @@
-		"react/jsx-handler-names": [1, {
+		"react/jsx-handler-names": ["warn", {
 			"eventHandlerPrefix": "handle",
 			"eventHandlerPropPrefix": "on"
 		}],
 		// Require a tab indentation in props.
-		"react/jsx-indent-props": [2, "tab"],
+		"react/jsx-indent-props": ["error", "tab"],
 		// Require tab indentations in all JSX components
-		"react/jsx-indent": [2, "tab"],
+		"react/jsx-indent": ["error", "tab"],
 		// Require key props to be used.
-		"react/jsx-key": 2,
+		"react/jsx-key": "error",
 		// Limit number of props on a single line in JSX to 1
-		"react/jsx-max-props-per-line":[2, {
-			"maximum": 1
+		"react/jsx-max-props-per-line":["error", {
+			"maximum": 1,
+			"when": "always"
 		}],
-		// Allow arrow functions and binding because "premature optimization is the root of all evil"
-		// or at least because it messes with HMR. (h/t Donald Knuth via Gabe Isman)
-		"react/jsx-no-bind": 0,
+		// Require binding to happen in the constructor.
+		"react/jsx-no-bind": ["error", {
+			"allowArrowFunctions": true
+		}],
+		// Don't allow comments inside JSX statements.
+		"react/jsx-no-comment-textnodes": "error",
 		// Don't allow duplicate props in the same component
-		"react/jsx-no-duplicate-props": 2,
+		"react/jsx-no-duplicate-props": "error",
 		// Don't allow unwrapped strings
-		"react/jsx-no-literals": 2,
+		"react/jsx-no-literals": "error",
+		// Don't allow unsafe target=_blanks, must include rel='noreferrer noopener' too.
+		"react/jsx-no-target-blank": "error",
 		// Don't allow undeclared variables in JSX
-		"react/jsx-no-undef": 2,
+		"react/jsx-no-undef": "error",
 		// PascalCase for all components
-		"react/jsx-pascal-case": 2,
-		// Don't require prop types to be sorted in any particular order
-		"react/jsx-sort-prop-types": 0,
-		// Don't require props to be sorted in any particular order.
-		"react/jsx-sort-props": 0,
+		"react/jsx-pascal-case": "error",
+		// Don't require props to be sorted alphabetically, but put reserved props (key, ref) first,
+		// and callbacks last.
+		"react/jsx-sort-props": ["error", {
+			"noSortAlphabetically": true,
+			"reservedFirst": true,
+			"callbacksLast": true,
+		}],
+		// Don't allow a space between / and >. Require a space before self closing.
+		// No space before opening.
+		"react/jsx-tag-spacing": ["error", {
+			"closingSlash": "never",
+			"beforeSelfClosing": "always",
+			"afterOpening": "never"
+		}],
 		// Allow react to be called and not cause errors.
-		"react/jsx-uses-react": 2,
+		"react/jsx-uses-react": "error",
 		// Allow JSX variables to be marked as used
-		"react/jsx-uses-vars": 2,
-		// Don't allow problematic JSX properties
+		"react/jsx-uses-vars": "error",
+		// Require parentheses around miltline JSX statements.
+		"react/jsx-wrap-multilines": "error",
+		// Warn if you're using array indexes as a key. It might be unavoidable, but use a unique id
+		// if possible.
+		"react/no-array-index-key": "warn",
```

Comment:
This might not be possible when things don't have unique IDs. There are plenty of plausible scenarios where we're not getting something with a UUID or such. We could create a random uuid, but that seems like overkill. In most scenarios, I imagine that the array key would be sufficient for what we're doing with it, but at least the developer will be warned and, hopefully, will act responsibly in looking up what's happening and alter accordingly.